bench: BenchmarkFuncExprTypeCheck fails on master

On 48065605ca2f44861b3d88c58b218acc28694ce4 I get the following:
```
yuzefovich@crlMBP-YN9GWF6WF7NTI1 cockroach % ./dev bench pkg/bench -f=BenchmarkFuncExprTypeCheck -v --stream-output
$ bazel test pkg/bench:all --test_arg -test.run=- --test_arg -test.bench=BenchmarkFuncExprTypeCheck --test_sharding_strategy=disabled --test_arg -test.v --crdb_test_off --test_output streamed
INFO: Invocation ID: b39f29a0-15bb-4ea5-932b-7423e43f80c6
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed 3 targets (1 packages loaded, 216 targets configured).
INFO: Found 2 targets and 1 test target...
I230221 17:41:05.799887 1 (gostd) rand.go:199  [T1] 1  random seed: 2662163393425416388
goos: darwin
goarch: arm64
BenchmarkFuncExprTypeCheck
    test_log_scope.go:161: test logs captured to: /Users/yuzefovich/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/43bfbdbe7df5c7b3e2a2cb71efa9af4e/logBenchmarkFuncExprTypeCheck242956656
    test_log_scope.go:79: use -show-logs to present logs inline
    bench_test.go:1443: error executing 'CREATE FUNCTION abs(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$': pq: cannot perform schema change on function abs(val: int) -> int because it is required by the database system
    panic.go:522: -- test log scope end --
test logs left over in: /Users/yuzefovich/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/43bfbdbe7df5c7b3e2a2cb71efa9af4e/logBenchmarkFuncExprTypeCheck242956656
--- FAIL: BenchmarkFuncExprTypeCheck
FAIL
I230221 17:41:06.168402 1 (gostd) testmain.go:136  [T1] 1  Test //pkg/bench:bench_test exited with error code 1
FAIL: //pkg/bench:bench_test (see /private/var/tmp/_bazel_yahor/e311b1c2454b7c301a6b73bddff450e7/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/testlogs/pkg/bench/bench_test/test.log)
INFO: Elapsed time: 24.825s, Critical Path: 24.33s
INFO: 35 processes: 1 internal, 34 darwin-sandbox.
INFO: Build completed, 1 test FAILED, 35 total actions
//pkg/bench:bench_test                                                   FAILED in 1.7s
  /private/var/tmp/_bazel_yahor/e311b1c2454b7c301a6b73bddff450e7/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/testlogs/pkg/bench/bench_test/test.log

INFO: Build completed, 1 test FAILED, 35 total actions
ERROR: exit status 3
```
I'll skip the failing stuff for now.

Jira issue: CRDB-24682

--- a/pkg/bench/bench_test.go
+++ b/pkg/bench/bench_test.go
@@ -1443,8 +1443,7 @@ func BenchmarkFuncExprTypeCheck(b *testing.B) {
 	sqlDB.ExecMultiple(b,
 		`CREATE SCHEMA sc1`,
 		`CREATE SCHEMA sc2`,
-		// TODO(chengxiong): uncomment this when #97400 is resolved.
-		//`CREATE FUNCTION abs(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$`,
+		`CREATE FUNCTION abs(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$`,
 		`CREATE FUNCTION sc1.udf(val INT) RETURNS INT CALLED ON NULL INPUT LANGUAGE SQL AS $$ SELECT val $$`,
 		`CREATE FUNCTION sc1.udf(val STRING) RETURNS STRING LANGUAGE SQL AS $$ SELECT val $$`,
 		`CREATE FUNCTION sc1.udf(val FLOAT) RETURNS FLOAT LANGUAGE SQL AS $$ SELECT val $$`,
@@ -1488,11 +1487,10 @@ func BenchmarkFuncExprTypeCheck(b *testing.B) {
 			name:    "builtin aggregate not called on null",
 			exprStr: "concat_agg(NULL)",
 		},
-		// TODO(chengxiong): uncomment this when #97400 is resolved.
-		//{
-		//	name:    "udf same name as builtin",
-		//	exprStr: "abs(123)",
-		//},
+		{
+			name:    "udf same name as builtin",
+			exprStr: "abs(123)",
+		},
 		{
 			name:    "udf across different schemas",
 			exprStr: "udf(123)",

--- a/pkg/sql/logictest/testdata/logic_test/udf
+++ b/pkg/sql/logictest/testdata/logic_test/udf
@@ -3133,3 +3133,22 @@ SELECT f_97130();
 
 statement ok
 SET search_path = $pre_search_path
+
+subtest regression_97400
+
+# Make sure that creating a UDF with builtin function name is ok.
+statement ok
+CREATE FUNCTION abs(val INT) RETURNS INT
+CALLED ON NULL INPUT
+LANGUAGE SQL
+AS $$ SELECT val+100 $$;
+
+query I
+SELECT abs(-1)
+----
+1
+
+query I
+SELECT public.abs(-1)
+----
+99

--- a/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_function.go
+++ b/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_function.go
@@ -32,8 +32,13 @@ func CreateFunction(b BuildCtx, n *tree.CreateFunction) {
 	dbElts, scElts := b.ResolvePrefix(n.FuncName.ObjectNamePrefix, privilege.CREATE)
 	_, _, sc := scpb.FindSchema(scElts)
 	_, _, db := scpb.FindDatabase(dbElts)
+	_, _, scName := scpb.FindNamespace(scElts)
+	_, _, dbname := scpb.FindNamespace(dbElts)
 
-	validateParameters(b, n, db.DatabaseID)
+	n.FuncName.SchemaName = tree.Name(scName.Name)
+	n.FuncName.CatalogName = tree.Name(dbname.Name)
+
+	validateParameters(n)
 
 	existingFn := b.ResolveUDF(
 		&tree.FuncObj{
@@ -142,7 +147,7 @@ func CreateFunction(b BuildCtx, n *tree.CreateFunction) {
 	b.Add(b.WrapFunctionBody(fnID, fnBodyStr, lang, refProvider))
 }
 
-func validateParameters(b BuildCtx, n *tree.CreateFunction, parentDBID descpb.ID) {
+func validateParameters(n *tree.CreateFunction) {
 	seen := make(map[tree.Name]struct{})
 	for _, param := range n.Params {
 		if param.Name != "" {

